### PR TITLE
Allow clients to set cache live time

### PR DESCRIPTION
**Pull request by pvg-alex**
Add interface method to change cache live time.

### CODE REVIEW

File: ironbank/src/main/java/com/quizup/store/LruDiskStore.java [hunk 1]
```diff
@@ -156,7 +159,8 @@ public boolean didExpire(String key)
             try
             {
                 long itemAge = System.currentTimeMillis() - itemInteractionTimestamps.get(key);
-                b = itemAge > CACHE_LIVE_TIME;
+                long cacheLiveTime = customCacheLiveTime > 0 ? customCacheLiveTime : DEFAULT_CACHE_LIVE_TIME;
+                b = itemAge > cacheLiveTime;
```
Comment (Paradoxia) [inline ironbank/src/main/java/com/quizup/store/LruDiskStore.java]: ```b``` is not very expressive

File: ironbank/src/main/java/com/quizup/store/LruDiskStore.java [hunk 2]
```diff
@@ -225,6 +229,11 @@ public boolean removeFromCache(String key, boolean notify)
 
     }
 
+    protected void setCacheLiveTime(long customCacheLiveTime)
```
Comment (Paradoxia) [inline ironbank/src/main/java/com/quizup/store/LruDiskStore.java]: Use java-doc to describe if parameter ```customCacheLiveTime``` is in millisec, seconds etc. Or name the variable to something like ```customCacheLiveTimeInMillisec```

File: ironbank/src/main/java/com/quizup/store/ironbank/IronBankCache.java [hunk 3]
```diff
@@ -42,6 +42,19 @@ protected IronBankCache(DiskCache diskCache, String key, Class<V> vClass)
...
     @Override
+    public void setCacheLiveTime(long cacheLiveTime)
+    {
+        super.setCacheLiveTime(cacheLiveTime);
+        expirable = true;
+    }
+
+    public void setDefaultCacheLiveTime()
+    {
+        super.setCacheLiveTime(-1);
```
Comment (Paradoxia) [inline ironbank/src/main/java/com/quizup/store/ironbank/IronBankCache.java]: Define a constant for ```-1```

File: ironbank/src/main/java/com/quizup/store/ironbank/IronBankCache.java [hunk 4]
```diff
@@ -42,6 +42,19 @@ protected IronBankCache(DiskCache diskCache, String key, Class<V> vClass)
     }
 
     @Override
+    public void setCacheLiveTime(long cacheLiveTime)
```
Comment (Paradoxia) [inline ironbank/src/main/java/com/quizup/store/ironbank/IronBankCache.java]: Java-doc the ```cacheLiveTime``` parameter to describe if it's in millisec, seconds etc. Or let the variable name express it (as mentioned above)

File: ironbank/src/main/java/com/quizup/store/LruDiskStore.java [hunk 5]
```diff
@@ -38,6 +38,8 @@
 
     private PublishSubject<String> removeSubject;
 
+    private long customCacheLiveTime = -1;
```
Comment (Paradoxia) [inline ironbank/src/main/java/com/quizup/store/LruDiskStore.java]: Make constant for ```-1```

File: ironbank/src/main/java/com/quizup/store/LruDiskStore.java (hunk 2 above)
Comment (pvg-alex) [inline ironbank/src/main/java/com/quizup/store/LruDiskStore.java]: Good point, I'll give a try to the `TimeUnit`

**Pull request merged**